Mathlib/Algebra/Order/Field/InjSurj.lean

```diff
@@ -3,58 +3,56 @@ Copyright (c) 2014 Robert Lewis. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Robert Lewis, Leonardo de Moura, Mario Carneiro, Floris van Doorn
 -/
-import Mathlib.Algebra.Order.Field.Defs
 import Mathlib.Algebra.Field.Basic
+import Mathlib.Algebra.Order.Field.Defs
 import Mathlib.Algebra.Order.Ring.InjSurj
 
 #align_import algebra.order.field.inj_surj from "leanprover-community/mathlib"@"ee0c179cd3c8a45aa5bffbf1b41d8dbede452865"
 
 /-!
-# Pulling back linearly ordered fields along injective maps.
-
+# Pulling back linearly ordered fields along injective maps
 -/
 
-
 open Function OrderDual
 
 variable {ι α β : Type*}
 
-namespace Function
+namespace Function.Injective
+variable [Zero β] [One β] [Add β] [Mul β] [Neg β] [Sub β] [Pow β ℕ] [SMul ℕ β]
+  [SMul ℤ β] [SMul ℚ β] [NatCast β] [IntCast β] [RatCast β] [Inv β] [Div β] [Pow β ℤ]
+  [Sup β] [Inf β] (f : β → α) (hf : Injective f)
 
--- See note [reducible non-instances]
 /-- Pullback a `LinearOrderedSemifield` under an injective map. -/
-@[reducible]
-def Injective.linearOrderedSemifield [LinearOrderedSemifield α] [Zero β] [One β] [Add β] [Mul β]
-    [Pow β ℕ] [SMul ℕ β] [NatCast β] [Inv β] [Div β] [Pow β ℤ] [Sup β] [Inf β] (f : β → α)
-    (hf : Injective f) (zero : f 0 = 0) (one : f 1 = 1) (add : ∀ x y, f (x + y) = f x + f y)
-    (mul : ∀ x y, f (x * y) = f x * f y) (inv : ∀ x, f x⁻¹ = (f x)⁻¹)
-    (div : ∀ x y, f (x / y) = f x / f y) (nsmul : ∀ (x) (n : ℕ), f (n • x) = n • f x)
+@[reducible] -- See note [reducible non-instances]
+def linearOrderedSemifield [LinearOrderedSemifield α] (zero : f 0 = 0) (one : f 1 = 1)
+    (add : ∀ x y, f (x + y) = f x + f y) (mul : ∀ x y, f (x * y) = f x * f y)
+    (inv : ∀ x, f x⁻¹ = (f x)⁻¹) (div : ∀ x y, f (x / y) = f x / f y)
+    (nsmul : ∀ (n : ℕ) (x), f (n • x) = n • f x)
     (npow : ∀ (x) (n : ℕ), f (x ^ n) = f x ^ n) (zpow : ∀ (x) (n : ℤ), f (x ^ n) = f x ^ n)
-    (nat_cast : ∀ n : ℕ, f n = n) (hsup : ∀ x y, f (x ⊔ y) = max (f x) (f y))
-    (hinf : ∀ x y, f (x ⊓ y) = min (f x) (f y)) : LinearOrderedSemifield β :=
-  { hf.linearOrderedSemiring f zero one add mul nsmul npow nat_cast hsup hinf,
-    hf.semifield f zero one add mul inv div nsmul npow zpow nat_cast with }
+    (natCast : ∀ n : ℕ, f n = n)
+    (hsup : ∀ x y, f (x ⊔ y) = max (f x) (f y)) (hinf : ∀ x y, f (x ⊓ y) = min (f x) (f y)) :
+    LinearOrderedSemifield β where
+  toLinearOrderedCommSemiring :=
```

Comment:
I think this `toX` style is perhaps no longer preferred now that the upstream bug is fixed
```suggestion
  __ :=
```
(the change to `linearOrderedCommSemiring` is a good one though)